windows/table.cpp

```diff
@@ -0,0 +1,275 @@
+#include "uipriv_windows.hpp"
+
+static void uiTableDestroy(uiControl *c);
+static void uiTableMinimumSize(uiWindowsControl *c, int *width, int *height);
+static BOOL onWM_NOTIFY(uiControl *c, HWND hwnd, NMHDR *nm, LRESULT *lResult);
+
+struct uiTable;
+
+struct uiTableModel {
+	uiTableModelHandler *mh;
+	std::vector<uiTable*> tables;
+};
+
+struct uiTableColumn {
+	uiTable *t;
+	WCHAR *name;
+	// don't really support parts (but this would part=>column mappings if we did)
+	int modelColumn;	// -1 = none
+};
+
+struct uiTable {
+	uiWindowsControl c;
+	uiTableModel *model;
+	HWND hwnd;
+	std::vector<uiTableColumn*> columns;
+};
+
+void *uiTableModelStrdup(const char *str)
+{
+	return strdup(str);
+}
+
+void *uiTableModelGiveColor(double r, double g, double b, double a)
+{
+	return 0;	// not implemented
+}
+
+uiTableModel *uiNewTableModel(uiTableModelHandler *mh)
+{
+	uiTableModel *m;
+
+	m = new uiTableModel();
+	m->mh = mh;
+	return m;
+}
+
+
+void uiFreeTableModel(uiTableModel *m)
+{
+	delete m;
+}
+
+void uiTableModelRowInserted(uiTableModel *m, int newIndex)
+{
+	LVITEM item;
+
+	item.mask = 0;
+	item.iItem = newIndex;
+	item.iSubItem = 0; //?
+	for (auto t : m->tables) {
+		ListView_InsertItem( t->hwnd, &item );
+	}
+}
+
+void uiTableModelRowChanged(uiTableModel *m, int index)
+{
+	for (auto t : m->tables) {
+		ListView_Update( t->hwnd, index );
+	}
+}
+
+void uiTableModelRowDeleted(uiTableModel *m, int oldIndex)
+{
+	for (auto t : m->tables) {
+		ListView_DeleteItem( t->hwnd, oldIndex );
+	}
+}
+
+void uiTableColumnAppendTextPart(uiTableColumn *c, int modelColumn, int expand)
+{
+	uiTable *t = c->t;
+	int lvIndex = 0;
+
+	if (c->modelColumn >=0) {
+		return; // multiple parts not implemented
+	}
+	c->modelColumn = modelColumn;
+
+	// work out appropriate listview index for the column
+	for (auto candidate : t->columns) {
+		if (candidate == c) {
+			break;
+		}
+		if (candidate->modelColumn >= 0) {
+			++lvIndex;
+		}
+	}
+	
+	LV_COLUMN lvc;
+	lvc.mask = LVCF_FMT | LVCF_WIDTH | LVCF_TEXT;	/* | LVCF_SUBITEM; */
+	lvc.fmt = LVCFMT_LEFT;
+	lvc.cx = 120;	// TODO
+	lvc.pszText = c->name;
+	ListView_InsertColumn(c->t->hwnd, lvIndex, &lvc);
+}
+
+void uiTableColumnAppendImagePart(uiTableColumn *c, int modelColumn, int expand)
+{
+	// not implemented
+}
+
+void uiTableColumnAppendButtonPart(uiTableColumn *c, int modelColumn, int expand)
+{
+	// not implemented
+}
+
+void uiTableColumnAppendCheckboxPart(uiTableColumn *c, int modelColumn, int expand)
+{
+	// not implemented
+}
+
+void uiTableColumnAppendProgressBarPart(uiTableColumn *c, int modelColumn, int expand)
+{
+	// not implemented
+}
+
+void uiTableColumnPartSetEditable(uiTableColumn *c, int part, int editable)
+{
+	// TODO
+}
+
+void uiTableColumnPartSetTextColor(uiTableColumn *c, int part, int modelColumn)
+{
+	// not implemented
+}
+
+// uiTable implementation
+
+uiWindowsControlAllDefaultsExceptDestroy(uiTable)
+
+uiTable *uiNewTable(uiTableModel *model)

```

Comment:
This should come after all the other uiTable methods, for consistency. This will also avoid the need to forward-declare the Destroy function as you did above.